ch_tools/monrun_checks/ch_backup.py

```diff
@@ -112,23 +112,30 @@ def _check_last_backup_not_failed(backups: List[Dict], crit_threshold: int) -> R
     """
     Check that the last backup is not failed. Its status must be 'created' or 'creating'.
     """
-    counter = 0
+    counter, userfault_counter = 0, 0
     for i, backup in enumerate(backups):
         state = backup["state"]
 
         if state == "created":
             break
 
-        if state == "failed" or (state == "creating" and i > 0):
-            counter += 1
+        if (state == "failed") or (state == "creating" and i > 0):
+            if "exception" not in backup or not _is_userfault_exception(
+                backup["exception"]
+            ):
+                counter += 1
+            else:
+                userfault_counter += 1
 
-    if counter == 0:
+    total_count = counter + userfault_counter
+
+    if total_count == 0:
         return Result(OK)
 
-    if counter == 1:
+    if total_count == 1:
         message = "Last backup failed"
     else:
-        message = f"Last {counter} backups failed"
+        message = f"Last {total_count} backups failed"
 
     status = CRIT if crit_threshold and counter >= crit_threshold else WARNING
     return Result(status, message)
```

Comment:
If we suppress crit due to user fault errors, it makes sense to add corresponding note to the result message. Something like `... (suppressed due to user fault errors)". There are similar functionality in `_suppress_if_required` function.